gen/enum_test.go

```diff
@@ -399,3 +400,107 @@ func TestEnumAccessors(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumLabelLegit(t *testing.T) {
+	tests := []struct {
+		enumItem  te.EnumWithLabel
+		wireValue string
+	}{
+		{te.EnumWithLabelUsername, "\"surname\""},
+		{te.EnumWithLabelPassword, "\"hashed_password\""},
+		{te.EnumWithLabelSalt, "\"salt\""},
+		{te.EnumWithLabelSugar, "\"sugar\""},
+	}
+	for _, tt := range tests {
+		t.Run("compare label:"+tt.wireValue, func(t *testing.T) {
+			// marshal
+			label := te.EnumWithLabel(tt.enumItem)
+			b, err := json.Marshal(label)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.wireValue, string(b))
+
+			// unmarshal
+			var expectedLabel te.EnumWithLabel
+			err = json.Unmarshal(b, &expectedLabel)
+			assert.Equal(t, expectedLabel, label)
```

Comment:
With assert.Equal, the call pattern is `assert.Equal(t, expected, got)`. Here, `label` is what we expect and `expectedLabel` is what we got (from deserializing). So this should be,

    assert.Equal(t, label, expectedLabel)

(We should fix naming.)